File: server/handlers/plan.go
package handlers

import (
	"encoding/json"
	"fitness-server/api"
	"net/http"
	"time"

	"github.com/go-chi/chi/v5"
	"github.com/google/uuid"
)

func GenerateFitnessPlan(w http.ResponseWriter, r *http.Request) {
	var request api.PlanGenerationRequest
	
	if err := json.NewDecoder(r.Body).Decode(&request); err != nil {
		http.Error(w, "Invalid JSON", http.StatusBadRequest)
		return
	}

	notesCoach := "Welcome to your personalized fitness plan!"
	sessionId := "session-1"
	samplePlan := api.Plan{
		PlanId:     uuid.New().String(),
		UserId:     request.UserId,
		Goal:       request.Goal,
		StartDate:  time.Now(),
		NotesCoach: &notesCoach,
		Weeks: []api.PlannedWeek{
			{
				WeekNumber: 1,
				StartDate:  time.Now(),
				Days: []api.PlannedDay{
					{
						Date:      time.Now(),
						SessionId: &sessionId,
					},
				},
			},
		},
		Sessions: []api.Session{
			{
				Id:    "session-1",
				Title: "Upper Body Strength",
				Blocks: []api.Block{
					{
						Type:                api.BlockTypeWarmup,
						Rounds:              1,
						RestSecBetweenRounds: 0,
						Items: []api.BlockStep{
							{
								Kind: api.BlockStepKindWarmup,
								WarmupStep: &api.WarmupStep{
									Description: "Dynamic arm swings",
									DurationSec: 300,
								},
							},
						},
					},
					{
						Type:                api.BlockTypeStrength,
						Rounds:              3,
						RestSecBetweenRounds: 90,
						Items: []api.BlockStep{
							{
								Kind: api.BlockStepKindExercisePrescription,
								ExercisePrescription: &api.ExercisePrescription{
									ExerciseName: "Push-ups",
									RepSpec: api.RepSpec{
										Kind: api.Reps,
										Reps: intPtr(10),
									},
									Intensity: api.Intensity{
										Mode: api.Rpe,
										Rpe:  float32Ptr(7.0),
									},
								},
							},
						},
					},
				},
			},
		},
	}

	w.Header().Set("Content-Type", "application/json")
	json.NewEncoder(w).Encode(samplePlan)
}

func GetUserPlan(w http.ResponseWriter, r *http.Request) {
	userID := chi.URLParam(r, "userID")
	
	if userID == "" {
		http.Error(w, "User ID required", http.StatusBadRequest)
		return
	}

	response := map[string]interface{}{
		"user_id": userID,
		"message": "Plan retrieval not yet implemented",
	}

	w.Header().Set("Content-Type", "application/json")
	json.NewEncoder(w).Encode(response)
}

func intPtr(i int) *int {
	return &i
}

func float32Ptr(f float32) *float32 {
	return &f
}

File: ascent/lib/models/plan_concepts/plan.dart
import '../../enums/goal.dart';
import 'planned_week.dart';
import 'session.dart';

class Plan {
  final String planId;
  final String userId;
  final Goal goal;
  final DateTime startDate;
  final List<PlannedWeek> weeks;          // calendar
  final List<Session> sessions;    // session list
  final String notesCoach;

  Plan({
    required this.planId,
    required this.userId,
    required this.goal,
    required this.startDate,
    List<PlannedWeek>? weeks,
    List<Session>? sessions,
    this.notesCoach = '',
  })  : 
        weeks = weeks ?? <PlannedWeek>[],
        sessions = sessions ?? <Session>[];

  factory Plan.fromJson(Map<String, dynamic> json) {
    return Plan(
      planId: json['plan_id'] as String,
      userId: json['user_id'] as String,
      goal: goalFromString(json['goal'] as String),
      startDate: _dateFromJson(json['start_date'] as String),
      weeks: (json['weeks'] as List<dynamic>? )?.map((e)=> PlannedWeek.fromJson(Map<String, dynamic>.from(e))).toList() ?? <PlannedWeek>[],
      sessions: (json['sessions'] as List<dynamic>?)?.map((e)=> Session.fromJson(Map<String, dynamic>.from(e))).toList() ?? <Session>[],
      notesCoach: (json['notes_coach'] as String?) ?? '',
    );
  }

  Map<String, dynamic> toJson() => {
    'plan_id': planId,
    'user_id': userId,
    'goal': goalToString(goal),
    'start_date': _dateToJson(startDate),
    'weeks': weeks.map((e)=> e.toJson()).toList(),
    'sessions': sessions.map((e)=> e.toJson()).toList(),
    'notes_coach': notesCoach,
  };
}

DateTime _dateFromJson(String value) => DateTime.parse(value);
String _dateToJson(DateTime value) => value.toIso8601String().split('T').first;